Remotion/ObjectBinding/Validation/BusinessObjectValidationResult.cs

```diff
@@ -48,29 +54,78 @@ private static BusinessObjectValidationFailure CreateBusinessObjectValidationFai
+                  validatedBusinessObject,
+                  validatedBusinessObjectProperty));
         }
 
         case ObjectValidationFailure objectValidationFailure
-            when validationFailure.ValidatedObject is IBusinessObject validatedBusinessObject:
+            when objectValidationFailure.ValidatedProperties.Count > 0:
         {
-          return BusinessObjectValidationFailure.CreateForBusinessObject(
-              validatedBusinessObject,
+          return CreateBusinessObjectValidationFailuresFromValidatedProperties(
+              objectValidationFailure.ValidatedProperties,
               objectValidationFailure.LocalizedValidationMessage);
         }
 
+        case ObjectValidationFailure objectValidationFailure
+            when validationFailure.ValidatedObject is IBusinessObject validatedBusinessObject:
+        {
+          return EnumerableUtility.Singleton(
+              BusinessObjectValidationFailure.CreateForBusinessObject(
+                  validatedBusinessObject,
+                  objectValidationFailure.LocalizedValidationMessage));
+        }
+
         default:
         {
-          return BusinessObjectValidationFailure.Create(validationFailure.LocalizedValidationMessage);
+          return EnumerableUtility.Singleton(BusinessObjectValidationFailure.Create(validationFailure.LocalizedValidationMessage));
         }
       }
     }
 
+    private static IEnumerable<BusinessObjectValidationFailure> CreateBusinessObjectValidationFailuresFromValidatedProperties (
+        IEnumerable<ValidatedProperty> validatedProperties,
+        string localizedValidationMessage)
+    {
+      var iterated = false;
+      foreach (var validatedProperty in validatedProperties)
+      {
+        var failure = CreateBusinessObjectValidationFailureFromValidatedProperty(validatedProperty, localizedValidationMessage);
+        if (failure == null)
+          continue;
+
+        yield return failure;
+        iterated = true;
+      }
+
+      if (!iterated)
+        yield return BusinessObjectValidationFailure.Create(localizedValidationMessage);
+    }
+
+    private static BusinessObjectValidationFailure? CreateBusinessObjectValidationFailureFromValidatedProperty (
+        ValidatedProperty validatedProperty,
+        string localizedValidationMessage)
+    {
+      if (validatedProperty.Object is not IBusinessObject validatedBusinessObject)
+        return null;
+
+      if (GetBusinessObjectProperty(validatedBusinessObject, validatedProperty.Property) is { } validatedBusinessObjectProperty)
+      {
+        return BusinessObjectValidationFailure.CreateForBusinessObjectProperty(
+            localizedValidationMessage,
+            validatedBusinessObject,
+            validatedBusinessObjectProperty);
+      }
+
+      return BusinessObjectValidationFailure.CreateForBusinessObject(
+          validatedBusinessObject,
+          localizedValidationMessage);
```

Comment:
Question: are those two equivalent (=symmetric) cases? if so, an IF-ELSE instead of an IF + separate "default" behavior would be better to provide code symmetry in the code.